src/index.cpp

```diff
@@ -1971,13 +2002,30 @@ template <typename IdType>
 std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search(const T *query, const size_t K, const uint32_t L,
                                                              IdType *indices, float *distances)
 {
+    IndexSearchContext<LabelT> context;
+    return search(query, K, L, indices, distances, context);
+}
+
+template <typename T, typename TagT, typename LabelT>
+template <typename IdType>
+std::pair<uint32_t, uint32_t> Index<T, TagT, LabelT>::search(const T *query, const size_t K, const uint32_t L,
+                                                             IdType *indices, float *distances,
+                                                             IndexSearchContext<LabelT> &context)
+{
+    if (context.UseFilter())
+    {
+        return search_with_filters(query, K, L, indices, distances, context);
+    }
+
     if (K > (uint64_t)L)
     {
```

Comment:
Why not simply increase L to K? It seems extreme to throw an exception for this.